convert2rhel/utils/__init__.py

```diff
@@ -1115,3 +1116,33 @@ def write_json_object_to_file(path, data, mode=0o600):
     with open(path, mode="w") as handler:
         os.chmod(path, mode)
         json.dump(data, handler, indent=4)
+
+
+def warn_deprecated_env(env_name):
```

Comment:
I would add unit tests for this. E.g. (the example also does the job and is tested, I wanted to save your time :))

```
diff --git a/convert2rhel/unit_tests/utils/utils_test.py b/convert2rhel/unit_tests/utils/utils_test.py
index 2a61212a..223ebe16 100644
--- a/convert2rhel/unit_tests/utils/utils_test.py
+++ b/convert2rhel/unit_tests/utils/utils_test.py
@@ -40,7 +40,7 @@ from six.moves import mock
 
 from convert2rhel import exceptions, systeminfo, toolopts, unit_tests, utils  # Imports unit_tests/__init__.py
 from convert2rhel.systeminfo import system_info
-from convert2rhel.unit_tests import RunCmdInPtyMocked, RunSubprocessMocked, is_rpm_based_os
+from convert2rhel.unit_tests import RunCmdInPtyMocked, RunSubprocessMocked, is_rpm_based_os, conftest
 
 
 DOWNLOADED_RPM_NVRA = "kernel-4.18.0-193.28.1.el8_2.x86_64"
@@ -1109,3 +1109,41 @@ class TestRemovePkgs:
 )
 def test_remove_epoch_from_yum_nevra_notation(pkg_nevra, nvra_without_epoch):
     assert utils._remove_epoch_from_yum_nevra_notation(pkg_nevra) == nvra_without_epoch
+
+
+@pytest.mark.parametrize(
+    ("env_name", "env_value", "tool_opts_name", "message"),
+    (
+        (
+            "CONVERT2RHEL_OUTDATED_PACKAGE_CHECK_SKIP",
+            True,
+            "outdated_package_check_skip",
+            "The environment variable CONVERT2RHEL_OUTDATED_PACKAGE_CHECK_SKIP is deprecated and is set to be removed on Convert2RHEL 2.4.0.\n"
+            "Please, use the configuration file instead.",
+        ),
+    ),
+)
+def test_warn_deprecated_env(global_tool_opts, monkeypatch, env_name, env_value, tool_opts_name, message, caplog):
+    """Test setting the value based on env variable and it's logging."""
+    monkeypatch.setattr(utils, "tool_opts", global_tool_opts)
+    monkeypatch.setenv(env_name, env_value)
+
+    utils.warn_deprecated_env(env_name)
+
+    assert getattr(global_tool_opts, tool_opts_name) == str(env_value)
+    assert caplog.records[-1].message == message
+
+
+def test_warn_deprecated_env_wrong_name(global_tool_opts, monkeypatch, caplog):
+    """Test when unsupported env variable is used, nothing is set."""
+    monkeypatch.setattr(utils, "tool_opts", global_tool_opts)
+
+    utils.warn_deprecated_env("UNSUPPORTED_ENV_VAR")
+
+    # Get tool_opts with default values
+    default_tool_opts = toolopts.ToolOpts()
+    default_tool_opts.initialize(config_sources=[conftest.CliConfigMock(), conftest.FileConfigMock()])
+
+    for item, value in global_tool_opts.__dict__.items():
+        assert default_tool_opts.__dict__[item] == value
+    assert caplog.text == ""
```

Reply:
Alright, thanks! 

I did not add any tests for it because it will be removed in the future, but I guess I can include this one and also mention that it will be removed.